controller.go

```diff
@@ -282,6 +328,17 @@ func (c *controller) addNetwork(n *network) error {
 		}
 	}
 
+	c.Lock()
+	// Load the ipam driver if not done already
+	_, ok = c.ipams[n.ipamType]
```

Comment:
Can we wrap the whole looking up an ipam driver in controller in a `n.ipam()` which will try to lookup ipam from `c.ipams[]`. If not found it will load the driver. This is a similar change I have made for network drivers. This makes the the whole driver resolution happen wherever we need to get hold of ipam driver and not just during the network creation time because we need to get hold of ipam in other nodes where the network wasn't created but just being used to connect.